### ISSUE TITLE
Support for Android 9’s AutoFill functionality in Browser

### ISSUE THREAD

**Issue by Moxville** [bromite/bromite#230]
**Is your feature request related to privacy?**
No

**Is there a patch available for this feature somewhere?**
Not Applicable

**Describe the solution you'd like**
Add support for password managers to Autofill in browser (Android P)
News Source:
https://blog.dashlane.com/android-p-autofill/
https://blog.lastpass.com/2018/10/android-pie-autofill-save-time-autofill-lastpass-android.html/
https://www.xda-developers.com/android-p-autofill-browsers/
https://www.androidauthority.com/autofill-automatically-work-browsers-android-p-844555/

**Describe alternatives you've considered**
Not Applicable

**Comment by frasanvel** [comment]
I don't think this functionality depends on a cloud service. It's already sopported in other browsers like Firefox or Edge on Android.

I think it may be related to Bromite not exposing the full URL when using third party integrations (for example, autocomplete logins with password manager keyboards).

**Comment by javiersantos** [comment]
I don't think this functionality depends on a cloud service. It's already sopported in other browsers like Firefox or Edge on Android.

I think it may be related to Bromite not exposing the full URL when using third party integrations (for example, autocomplete logins with password manager keyboards).

**Comment by javiersantos** [comment]
Response from Enpass:

_"The problem is with the browser which is not revealing the URL being browsed and instead of that passing-on its own package name to Enpass. Actually, the behavior should be like Chrome which passes-on the exact URL to Enpass to save for future autofill operations. For better autofilling experience we recommend to use Chrome, Firefox-Focus or Edge  works well with autofill by Android framework as well as through Notifications."_

https://discussion.enpass.io/index.php?/topic/5563-wrong-autofill-information-gets-added-automatically/&do=findComment&comment=12006

**Comment by csagan5** [comment]
@javiersantos the only functionality disabled in Bromite is the interaction with Google APIs (it would not work regardless because a valid service account would be needed, and if it would work it would not be enabled in Bromite as it is now): https://github.com/bromite/bromite/blob/abfa65cd1751916188bc458bd7327ae46e367cfa/build/patches/autofill-disable-autofill-download-manager.patch

As you can see in that patch the part that would upload data is disabled.

For the rest, autofill is simply disabled by default, so you can enable it and test (I assume you have already enabled it before testing?).



> "The problem is with the browser which is not revealing the URL being browsed and instead of that passing-on its own package name to Enpass. Actually, the behavior should be like Chrome which passes-on the exact URL to Enpass to save for future autofill operations. For better autofilling experience we recommend to use Chrome, Firefox-Focus or Edge works well with autofill by Android framework as well as through Notifications."

@javiersantos when you submit a new issue (not the case when commenting on an existing issue like here) there is a section asking to test with the corresponding Chromium version. This has obviously not been done here, because you/the Enpass developer would verify that Chromium has exactly the same behaviour.
